### PR TITLE
nrf_rpc: made private bt serialization and proxy cb API the public NRF RPC API

### PR DESCRIPTION
**Pull request by alxelax**
PR includes:

- Implementation of `bt_conn_cb_unregister`. New BLE Host API was added in 2.7.0 but not serialized yet.
- Moving Bluetooth RPC serialization and proxy callback API as the public NRF RPC API

### CODE REVIEW

File: subsys/nrf_rpc/CMakeLists.txt [hunk 1]
```diff
@@ -8,5 +8,9 @@ zephyr_include_directories(include)
 
 zephyr_library()
 
-zephyr_library_sources(nrf_rpc_os.c)
+zephyr_library_sources(
+	nrf_rpc_os.c
+	nrf_rpc_serialize.c
```
Comment (KAGA164) [inline subsys/nrf_rpc/CMakeLists.txt:13]: nrf_rpc_serialize.c and nrf_rpc_cbkproxy.c could be compiled based on additional Kconfig options like NRF_RPC_SERIALIZE_API NRF_RPC_CALLBACK_PROXY

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -453,7 +453,7 @@ Libraries for NFC
 nRF RPC libraries
 -----------------
 
-|no_changes_yet_note|
+* Converted internal Bluetooth serialization API and Bluetooth callback proxy API into the public NRF RPC API.
```
Comment (annwoj) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:456]:
```suggestion
* Updated the internal Bluetooth serialization API and Bluetooth callback proxy API to become part of the public NRF RPC API.
```
preferable wording for the release notes

File: include/nrf_rpc/nrf_rpc_cbkproxy.h [hunk 3]
```diff
@@ -6,49 +6,56 @@
...
- *                      if callback does not take any parameres.
- * @param callback_arguments
- *                      List of arguments with the same names as in
- *                      ''handler_parameters'' enclosed in parentheses,
- *                      e.g. ''(buf, len)''. This macro parameter is optional
- *                      and must be skipped if callback does not take any
- *                      parameres.
+ * @param ... - variadic number of parameters:
+ *
+ * handler_name - name of the handler function that can ne passed to
```
Comment (annwoj) [inline include/nrf_rpc/nrf_rpc_cbkproxy.h:33]:
```suggestion
 * handler_name - name of the handler function that can be passed to the
```

File: include/nrf_rpc/nrf_rpc_cbkproxy.h [hunk 4]
```diff
@@ -6,49 +6,56 @@
...
+ * appended to the end: ''uint32_t callback_slot''.
+ *
+ * return_type - optional return type of the callback. It must be
+ * convertable to ''uint32_t'' without lost of information.
+ * If it is ''void'' this macro parameter must be skipped.
+ *
+ * handler_parameters - list of parameters with its types enclosed in
+ * parentheses, e.g. ''(uint8_t *buf, size_t len)''.
+ * This macro parameter is optional and must be skipped
+ * if callback does not take any parameres.
```
Comment (annwoj) [inline include/nrf_rpc/nrf_rpc_cbkproxy.h:51]:
```suggestion
 * This macro parameter is optional and must be skipped
 * if the callback does not take any parameters.
```

File: include/nrf_rpc/nrf_rpc_cbkproxy.h [hunk 5]
```diff
@@ -6,49 +6,56 @@
...
+ * callback_name - callback that will be called by this handler. Best
+ * approach is to declare it as 'static inline' to make
+ * sure that handler and callback will be combined into
+ * one function during the optimization. This callback
+ * has the same parameters as the original one (provided
+ * in ''handler_parameters'') with following parameter
+ * appended to the end: ''uint32_t callback_slot''.
+ *
+ * return_type - optional return type of the callback. It must be
+ * convertable to ''uint32_t'' without lost of information.
```
Comment (annwoj) [inline include/nrf_rpc/nrf_rpc_cbkproxy.h:45]:
```suggestion
 * convertible to ''uint32_t'' without a loss of information.
```

File: include/nrf_rpc/nrf_rpc_cbkproxy.h [hunk 6]
```diff
@@ -6,49 +6,56 @@
...
+ *
+ * handler_parameters - list of parameters with its types enclosed in
+ * parentheses, e.g. ''(uint8_t *buf, size_t len)''.
+ * This macro parameter is optional and must be skipped
+ * if callback does not take any parameres.
+ *
+ * callback_arguments - list of arguments with the same names as in
+ * ''handler_parameters'' enclosed in parentheses,
+ * e.g. ''(buf, len)''. This macro parameter is optional
+ * and must be skipped if callback does not take any parameres.
```
Comment (annwoj) [inline include/nrf_rpc/nrf_rpc_cbkproxy.h:56]:
```suggestion
 * and must be skipped if callback does not take any parameters.
```

File: include/nrf_rpc/nrf_rpc_cbkproxy.h [hunk 7]
```diff
@@ -6,49 +6,56 @@
...
+ *
+ * handler_name - name of the handler function that can ne passed to
+ *  @ref nrf_rpc_cbkproxy_out_get() function.
+ *
+ * callback_name - callback that will be called by this handler. Best
+ * approach is to declare it as 'static inline' to make
+ * sure that handler and callback will be combined into
+ * one function during the optimization. This callback
+ * has the same parameters as the original one (provided
+ * in ''handler_parameters'') with following parameter
```
Comment (annwoj) [inline include/nrf_rpc/nrf_rpc_cbkproxy.h:41]:
```suggestion
 * in ''handler_parameters'') with the following parameter
```

File: include/nrf_rpc/nrf_rpc_cbkproxy.h [hunk 8]
```diff
@@ -6,49 +6,56 @@
...
+ * appended to the end: ''uint32_t callback_slot''.
+ *
+ * return_type - optional return type of the callback. It must be
+ * convertable to ''uint32_t'' without lost of information.
+ * If it is ''void'' this macro parameter must be skipped.
+ *
+ * handler_parameters - list of parameters with its types enclosed in
+ * parentheses, e.g. ''(uint8_t *buf, size_t len)''.
```
Comment (annwoj) [inline include/nrf_rpc/nrf_rpc_cbkproxy.h:49]:
```suggestion
 * parentheses, for example, ''(uint8_t *buf, size_t len)''.
```

File: include/nrf_rpc/nrf_rpc_cbkproxy.h [hunk 9]
```diff
@@ -6,49 +6,56 @@
...
+ * This macro parameter is optional and must be skipped
+ * if callback does not take any parameres.
+ *
+ * callback_arguments - list of arguments with the same names as in
+ * ''handler_parameters'' enclosed in parentheses,
+ * e.g. ''(buf, len)''. This macro parameter is optional
```
Comment (annwoj) [inline include/nrf_rpc/nrf_rpc_cbkproxy.h:55]:
```suggestion
 * for example, ''(buf, len)''. This macro parameter is optional
```

**Review by annwoj — COMMENTED** [review]
adding few small suggestions

**Review by annwoj — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]

**Review by bpienk — APPROVED** [review]

**Pull request merged**